test/README.md

```diff
@@ -0,0 +1,154 @@
+
+The servers are linked as follows: 
+
+* Server 1 
+  * -> Server 2  
+  * -> Server 3
+* Server 2 
+  * -> Server 1
+* Server 3 
+  * -> Server 2
+  * -> Server 1
+* Express app: (if you need to emulate any external endpoint, use the express app)  
+
+Server 1 runs on port `55551`, Server 2 on port `55552` and Server 3 on port `55553`.
+
+> If you have the need to increase the number of servers running, it is possible, but please discuss with the team before you go in that path.
+
+
+#### Adding a new block
+
+To add a new feature you need to export the feature as a function that take as an argument any of the servers you want to interact. 
+
+
+```js
+// newFeature.js
+
+export default function(server) {
+
+  describe('package access control', () => {
+    test('should ...', (done) => {
+      done();
+    });
+  });
+  
+}
+```
+
+Then, import the feature and run the function within the main `describe` block.  
+
+```js
+// index.spec.js
+
+import newFeature from './newFeature';
+
+describe('functional test verdaccio', function() {
+  // test are fast, but do not change this time out, 10 seconds should be more than enough
+  jest.setTimeout(10000);
+  // servers are accessed vi a global jest state.
+  const server1: IServerBridge = global.__SERVERS__[0];
+  const server2: IServerBridge = global.__SERVERS__[1];
+  const server3: IServerBridge = global.__SERVERS__[2];
+  const app = global.__WEB_SERVER__.app;
+
+  // include as much servers you need
+  newFeature(server1, server2, server3);
+});
+```
+
+Functional test run over ones single file, thus, it is not possible at this stage to run test individually.
+
+### E2E Test
+
+Verdaccio includes an User Interface and must be tested. We use End-to-End testing to run some smock test against the web API using the UI Theme 
+include by default.
+
+```bash
+    yarn lint && yarn test:all
+```
+
+The test does not have aim to test the integrity of the page, mostly, ensure the basic functionality still works. If you add or modify 
+a UI feature, the tests must be updated.
+
+> The tests rely on CSS classes naming convention, so, it is required some sort of coordination with the **verdaccio/ui** project.
+
+We uses `puppeteer`, you can find more information about how to use it in their website. 
+
+## Before commit
+
+We recommend run your tests and linters before commit.   
+
+```bash
+   yarn lint && yarn test:all
+```
+
+
+You can find more in our [guide about run and debugging test](https://github.com/verdaccio/verdaccio/wiki/Running-and-Debugging-tests#running-the-test).
+
+## Continuous Integration
+
+Verdaccio uses [CircleCI](https://circleci.com/gh/verdaccio) as main Continuous Integration tool. We run the test against the most common 
+Node.js versions available, LTS and the latest release. Before the PR is being merged, all check must be green.   
```

Comment:
```suggestion
```

Reply:
Why we remove this? I was not 100% happy when I wrote it, but now :) I'm curious.

Reply:
I merged it with the paragraph and removed it from being it's own single line. That line of text is still there :-)

Reply:
Ah ok 👍 I saw it.